### ISSUE TITLE
Problem with reading contents of calcite-list and calcite-popover in calcite-dialog

### ISSUE THREAD

**Issue by cont-amibabkin** [Esri/calcite-components#12421]
**monday.com sync:** #18005841626

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

Instead of calcite-select we use a combination of `calcite-button` with `calcite-popover` and `calcite-list` inside for more flexible customization. We encountered the fact that when using our custom select inside a `calcite-dialog`, the screen reader reads out the entire contents of the custom select when it is opened.

### Actual Behavior

When you open a popover with a list, the following text is read out:

![Image](https://github.com/user-attachments/assets/529fde71-1f10-4cc8-83db-f4d9951de81f)

### Expected Behavior

We would like the screen reader to read the text when opening our custom select in a `calcite-dialog`, just like when opening our custom select that is not inside a `calcite-dialog`.

![Image](https://github.com/user-attachments/assets/49328a98-326a-4cd5-a746-3cabbf16ed6c)

Notice how the screen reader reads out the text when we use a custom dialog.

![Image](https://github.com/user-attachments/assets/82cedd42-eb58-4c63-9bce-aa46941f3227)

### Reproduction Sample

Main example
https://codepen.io/Andrey-B-the-animator/pen/pvJGBxR

Custom dialog with calcite-popover and calcite-list
https://codepen.io/Andrey-B-the-animator/pen/XJbQKVL

### Reproduction Steps

1. Enable a screen reader.
2. Navigate to the Open Dialog button and activate it.
3. Navigate to the Layer information button in dialog, activate it and notice how the elements are declared.

### Reproduction Version

3.0.3

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by HauTruong146** [comment]
Hi calcite team, just want to follow up with this issue. I included a screenshot of a dialog that we are using in our application. The outer action (button in the screenshot) will open a popover, which has a outer wrap label that contains a div of a button and another popover. 

With the current implementation, since the popover has aria-live="polite", when the outer action is clicked, the outer popover shows and will announce to the screen reader the content. But it doesn't announce the label (inner label in this case) on first launch of the outer popover. Is there a work around for the screen reader to announce the label?

I tried attributes like `aria-labelledby, for` but neither work. 





<img width="1012" height="711" alt="Image" src="https://github.com/user-attachments/assets/825693a1-75a8-4217-bfe5-d4a0512cc49d" />


```
<html>

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
    <script type="module" src=https://js.arcgis.com/calcite-components/3.3.3/calcite.esm.js></script>
    <style>
        p {
            padding: 0.75rem;
            margin: 0;
        }
    </style>
</head>

<body>
    <calcite-popover heading="Outer popover" label="outer popover" reference-element="outer-action" closable>
        <calcite-label>
          inner label
          <div>
            <calcite-button id="inner-button" aria-controls="inner-popover">inner button</calcite-button>
            <calcite-popover id="inner-popover" reference-element="inner-button">some text</calcite-popover>
          </div>
        </calcite-label>
    </calcite-popover>
    <calcite-button id="outer-action">Layer information</calcite-button>
</body>

</html>
```

**Comment by HauTruong146** [comment]
Hi @geospatialem, I also found a related issue with keyboard navigation when putting a list inside a popover.

    <calcite-popover label="This is a new feature" reference-element="popover-button">
        <calcite-list label="Park features">
        <calcite-list-item label="Hiking trails" description="Designated routes for hikers to use."
            value="hiking-trails">
            <calcite-action slot="actions-end" icon="layer" text="Trails layer"></calcite-action>
        </calcite-list-item>
        <calcite-list-item label="Waterfalls" description="Vertical drops from a river." value="waterfalls">
            <calcite-action slot="actions-end" icon="layer" text="Waterfalls layer"></calcite-action>
        </calcite-list-item>
        <calcite-list-item label="Rivers" description="Large naturally flowing watercourses." value="rivers">
            <calcite-action slot="actions-end" icon="layer" text="Rivers layer"></calcite-action>
        </calcite-list-item>
    </calcite-list>
    </calcite-popover>
    <calcite-button appearance="outline-fill" icon-start="pencil" id="popover-button">
        Leave us a note
    </calcite-button>


<img width="1064" height="1035" alt="Image" src="https://github.com/user-attachments/assets/9b91390d-de88-4de3-a0a6-ee2a8d0f0f51" />


On window, if the focus is on the calcite list item, it requires pressing ESC key twice to close the popover. But when the focus is on the icon, only one press is needed. 

A work around can be done by pressing "Shift + Esc" to close the popup in one press.

**Comment by dom-jones-1** [comment]
I'm seeing the same issue with the calcite-list screen reader text changing when it is created more than once. The first time it will read out the below (which I think is the desired outcome)
```
Park features  table
Hiking trails  not selected  1 of 3  level 1
```
Then any subsequent time it will just read out
```
Total items: 3. 
1. Hiking trails 2. Waterfalls 3. Rivers
```

I have created a sample Codepen [here](https://codepen.io/editor/dom-jones-1/pen/01a012da-322a-7ec7-99e5-9477cdf904ed). I can kind of get around it by waiting for the list to finish loading and focusing on it, which works well to interrupt NVDA announcing the second one and announce the desired one again, but it is kind of hacky and doesn't work well with VoiceOver.

You can see the first time the list is shown, the first item gets focused and has a border, but subsequent times that does not happen.

**Comment by ddred11** [comment]
@geospatialem Hello! Wondering if this issue can be moved up in priority?  This is a blocker for the BA team's push to provide a11y support to the BA report player by March.

**Comment by geospatialem** [comment]
Hi folks, quick update on this one - been doing some research and this may be a limitation of browser support and shadow DOM. Will investigate further WRT the spike effort and update folks on the effort.
